src/App.jsx

```diff
@@ -14,33 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+const REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPrepearedGoods(goods, { sortfield, reverse }) {
+  let prepearedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortfield) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortfield) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse === REVERSE) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortfield, setSortfield] = useState('');
+  const [reverse, setReverse] = useState('');
+  const visiblegoods = getPrepearedGoods(goodsFromServer, {
+    sortfield,
+    reverse,
+  });
+  const setreverseButton = () =>
+    reverse === REVERSE ? setReverse('') : setReverse(REVERSE);
```

Comment:
```suggestion
  const setreverseButton = () => setIsReversed(prevIsReversed => !prevIsReversed);
```
here just change prev state on opposite.